From f9886ac8eb8b017c61fd5cc45e9b694507371d86 Mon Sep 17 00:00:00 2001
From: Peter Hutterer <peter.hutterer@who-t.net>
Date: Tue, 19 Jul 2011 11:42:12 +1000
Subject: [PATCH 1/6] Don't free anything on PreInit failure, let UnInit take
 care of it

Avoid double free's by the server when it calls UnInit.

Signed-off-by: Peter Hutterer <peter.hutterer@who-t.net>
---
 src/xf86Fpit.c |    6 ------
 1 files changed, 0 insertions(+), 6 deletions(-)

diff --git a/src/xf86Fpit.c b/src/xf86Fpit.c
index fa4ace7..e0f316b 100644
--- a/src/xf86Fpit.c
+++ b/src/xf86Fpit.c
@@ -568,12 +568,6 @@ static int xf86FpitInit(InputDriverPtr drv, InputInfoPtr pInfo, int flags)
 	str = xf86FindOptionValue(pInfo->options, "Device");
 	if (!str) {
 		xf86Msg(X_ERROR, "%s: No Device specified in FPIT module config.\n", pInfo->name);
-		if (priv) {
-			if (priv->fpitDev) {
-				free(priv->fpitDev);
-			}
-			free(priv);
-		}
 		return BadValue;
 	}
 	priv->fpitDev = strdup(str);
-- 
1.7.6

